src/ibm/hypervisor-network-mgr-src/hyp_ethernet_interface.cpp

```diff
@@ -681,43 +794,131 @@ ObjectPath HypEthInterface::ip(HypIP::Protocol protType, std::string ipaddress,
 HypEthernetIntf::DHCPConf
     HypEthInterface::dhcpEnabled(HypEthernetIntf::DHCPConf value)
 {
-    if (value == HypEthernetIntf::dhcpEnabled())
+    HypEthernetIntf::DHCPConf oldValue = HypEthernetIntf::dhcpEnabled();
+
+    if (value == oldValue)
     {
         return value;
     }
 
-    HypEthernetIntf::dhcpEnabled(value);
-
     if (value != HypEthernetIntf::DHCPConf::none)
     {
+        bool v4Enabled = false;
+        bool v6Enabled = false;
+        bool slaacEnabled = false;
+        HypEthernetIntf::DHCPConf newValue;
+
+        if (value == HypEthernetIntf::DHCPConf::v4)
+        {
+            if (oldValue == HypEthernetIntf::DHCPConf::none)
+            {
+                newValue = value;
+                v4Enabled = true;
+            }
+            else if ((oldValue == HypEthernetIntf::DHCPConf::v6stateless) ||
+                     (oldValue == HypEthernetIntf::DHCPConf::v4v6stateless))
+            {
+                newValue = HypEthernetIntf::DHCPConf::v4v6stateless;
+                v4Enabled = true;
+                slaacEnabled = true;
+            }
+            else
+            {
+                newValue = HypEthernetIntf::DHCPConf::both;
+                v4Enabled = true;
+                v6Enabled = true;
+            }
+        }
+        else if (value == HypEthernetIntf::DHCPConf::v6)
+        {
+            if (oldValue == HypEthernetIntf::DHCPConf::none)
+            {
+                newValue = value;
+                v6Enabled = true;
+            }
+            else
```

Comment:
You should check for other values also. 
If old value is v6stateless, then set newValue to v6
If old value is v6, then keep new value as v6 only

Similarly other else-if conditions needs to be updated.

Reply:
+1